### DISCUSSION TITLE
Hover over doesn't work after first merge key if field exists in anchor definition

### DISCUSSION THREAD

**Discussion by qzcx** [redhat-developer/vscode-yaml#946] [General]
I created a json-schema for an internal tool, but there seems to be issues around using anchors to provide default values while overriding others. 

The test_base_anchor anchor is valid, and I can "go to definition" on it.
Fields specified before ('when') the anchor merge show the hover text.
Fields specified after ('desc') the anchor merge don't show. 
If I remove 'desc' from the test_name1 entry, then hover over for it works under test_name2.

Is this a bug, unsupported, or am I specifying something wrong in my schema?
Any advice?

```
test_name1: &test_base_anchor
  when: [blah11, blah12] 
  desc: Some description of test1

test_name2: #Hover works here
  when: [blah21, blah22] #hover over 'when' works here
  <<: *test_base_anchor
  desc: Some description of test2 #Hover over 'desc' doesn't work here
```

My json schema looks like this: 

```
{
  "type": "object",
  "patternProperties": {
    "^.*$": {
      "type": "object",
      "description": "Hover over for Test Name",
      "properties": {
        "when": {
          "type": "array",
          "items": {
            "type": "string"
          },
          "description": "Hover over description for when"
        },
        "desc": {
          "type": "string",
          "description": "hover over description for desc"
        }
    }
  }
}
```

Note, I've noticed that this syntax in a bit of limbo where only some tools support it based on the discussion in this stacked overflow. https://stackoverflow.com/questions/41063361/what-is-the-double-left-arrow-syntax-in-yaml-called-and-wheres-it-specifi

**Comment by FALLAI-Denis** [comment]
Hi,

I think merge key was introduced by and only supported by YAML 1.1 and is not supported / deprecated for YAML 1.2. 

Do you try to fix yaml.yamlVersion to 1.1?

See also https://github.com/hilleer/vscode-yaml-plus-json/issues/38

Regards.

**Reply by qzcx** [reply]
Yes, I did try switching yaml.yamlVersion to 1.1 and saw no difference.

Thanks for the link. That issue describes the full behavior I'd expect. 

Even if validation didn't work, this working for hover over documentation would be great since it would be better my poor implementation in my internal vscode extension.
